### ISSUE TITLE
Configure with no-stdio on Windows generates makefile that doesn't work with jom

### ISSUE THREAD

**Issue by axnsan12** [openssl/openssl#16014]
While building openssl from source, on the 1.1.1k tag (commit e2e09d9fba), using the following command:

`perl Configure VC-WIN64A no-tests no-stdio`

an invalid makefile is generated, specifically (around line 197):

```
...
clean: libclean
    
    -del /Q /F $(ENGINES)
    -del /Q /F $(SCRIPTS)
    ...
```

Notice the empty line. Compare with output when configuring without `no-stdio`:

```
...
clean: libclean
    -del /Q /F apps\openssl.exe
    -del /Q /F $(ENGINES)
    -del /Q /F $(SCRIPTS)
    ...
```

This causes a syntax error at line 197 when running the clean command.

I'm also attaching the [configdata dump](https://github.com/openssl/openssl/files/6774752/dump.txt), in case it helps.

EDIT: right after opening this issue I realised the problem is caused by me using [jom](https://wiki.qt.io/Jom) to run the build step. Using nmake works just fine as it seems to ignore the empty line. So I guess this decreases the severity of the issue somewhat.

**Comment by levitte** [comment]
Would you mind checking if this patch makes a difference?

```
diff --git a/Configurations/windows-makefile.tmpl b/Configurations/windows-makefile.tmpl
index 9351149fe8..ec0fd94641 100644
--- a/Configurations/windows-makefile.tmpl
+++ b/Configurations/windows-makefile.tmpl
@@ -362,7 +362,7 @@ libclean:
        -del /Q /F $(LIBS) libcrypto.* libssl.* ossl_static.pdb
 
 clean: libclean
-       {- join("\n\t", map { "-del /Q /F $_" } @PROGRAMS) -}
+       {- join("\t", map { "-del /Q /F $_\n" } @PROGRAMS) || "\@rem\n" -}
        -del /Q /F $(ENGINES)
        -del /Q /F $(SCRIPTS)
        -del /Q /F $(GENERATED_MANDATORY)
```

**Comment by levitte** [comment]
(that's not a complete patch, there are other spots like this, but those are conditional on certain configuration options)

**Comment by axnsan12** [comment]
Sorry for taking so long to respond, I tried the patch above and it does *not* work, it still leaves a blank line after the @rem line. Removing the \n (so just ` || "\@rem"`), the patch works properly.


```
diff --git a/Configurations/windows-makefile.tmpl b/Configurations/windows-makefile.tmpl
index 9351149fe8..38e5ec1594 100644
--- a/Configurations/windows-makefile.tmpl
+++ b/Configurations/windows-makefile.tmpl
@@ -362,7 +362,7 @@ libclean:
        -del /Q /F $(LIBS) libcrypto.* libssl.* ossl_static.pdb

 clean: libclean
-       {- join("\n\t", map { "-del /Q /F $_" } @PROGRAMS) -}
+    {- join("\t", map { "-del /Q /F $_\n" } @PROGRAMS) || "\@rem" -}
        -del /Q /F $(ENGINES)
        -del /Q /F $(SCRIPTS)
        -del /Q /F $(GENERATED_MANDATORY)
```

**Comment by levitte** [comment]
I found more places that could give these sorts of empty lines.

Fix for 1.1.1 in #16047.
Fix for 3.0 (master) in #16048

**Comment by axnsan12** [comment]
Thanks for the quick resolution, much appreciated!
